### PR TITLE
change creation and expiration to timestamps

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: src/runtime/entity.ts [hunk 1]
```diff
@@ -369,7 +368,7 @@ export abstract class Entity implements Storable {
   }
 
   static identify(entity: Entity, identifier: string, storageKey: string, creationTimestamp?: string) {
-    getInternals(entity).identify(identifier, storageKey, creationTimestamp);
+    getInternals(entity).identify(identifier, storageKey, new Date(Number(creationTimestamp)));
```
Comment (galganif) [inline src/runtime/entity.ts:371]: will this work is creationTimestamp is null? or maybe we want to make the parameter non nullable?

File: src/runtime/entity.ts [hunk 2]
```diff
@@ -231,10 +230,10 @@ class EntityInternals {
       rawData: this.dataClone()
     };
     if (this.hasCreationTimestamp()) {
-      serializedEntity.creationTimestamp = this.creationTimestamp;
+      serializedEntity.creationTimestamp = this.creationTimestamp.getTime().toString();
     }
     if (this.hasExpirationTimestamp()) {
-      serializedEntity.expirationTimestamp = this.expirationTimestamp;
+      serializedEntity.expirationTimestamp = this.expirationTimestamp.getTime().toString();
```
Comment (galganif) [inline src/runtime/entity.ts:236]: why not store it as a number instead of string?

File: src/runtime/reference.ts [hunk 3]
```diff
@@ -25,30 +25,37 @@ export type SerializedReference = {
   entityStorageKey: string;
   // TODO(#4861): creationTimestamp shouldn't be optional.
   creationTimestamp?: string;
-  // TODO(mmandlis): add expiration time here as well.
+  expirationTimestamp?: string;
```
Comment (galganif) [inline src/runtime/reference.ts:28]: can these be Dates as well? were you planning on changing these later?

File: src/runtime/reference.ts [hunk 4]
```diff
@@ -101,6 +113,7 @@ export abstract class ClientReference extends Reference {
       {
         id: Entity.id(entity),
         creationTimestamp: Entity.creationTimestamp(entity),
+        expirationTimestamp: Entity.creationTimestamp(entity),
```
Comment (galganif) [inline src/runtime/reference.ts:116]: don't we need expirationTimestamp(entity)?

File: src/runtime/entity.ts (hunk 1 above)
Comment (mmandlis) [reply src/runtime/entity.ts]: added a null check

File: src/runtime/entity.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/entity.ts]: changed to number, thanks

File: src/runtime/reference.ts (hunk 3 above)
Comment (mmandlis) [reply src/runtime/reference.ts]: Date isn't serializable, but changed to number instead.

File: src/runtime/reference.ts (hunk 4 above)
Comment (mmandlis) [reply src/runtime/reference.ts]: good catch, updated, thanks.

**Comment by mmandlis** [comment]
manual copybara passed cl/305262555, merging.

**Pull request merged**